client/alloc_runner.go

```diff
@@ -577,7 +577,9 @@ func (r *AllocRunner) Alloc() *structs.Allocation {
 	// The health has been set
 	if r.allocHealth != nil {
 		if alloc.DeploymentStatus == nil {
-			alloc.DeploymentStatus = &structs.AllocDeploymentStatus{}
+			alloc.DeploymentStatus = &structs.AllocDeploymentStatus{
+				Timestamp: time.Now(),
```

Comment:
Perhaps add a comment explaining why a timestamp in an empty DeploymentStatus struct is needed (without context, it can be confusing)